src/App.tsx

```diff
@@ -15,48 +24,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getPreparedItems(
+  { sortField, isReversed } : Methods,
+) {
+  const sortedGoods = [...goodsFromServer];
+
+  if (sortField) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case alphabeticallySort:
+          return good1.localeCompare(good2);
+
+        case lengthSort:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<string>('');
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const visibleGoods = getPreparedItems(
+    { sortField, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== alphabeticallySort,
+          })}
+          onClick={() => setSortField(alphabeticallySort)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== lengthSort,
+          })}
+          onClick={() => setSortField(lengthSort)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => {
+            setIsReversed(prev => !prev);
+          }}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField || isReversed) && (
+
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField('');
+              setIsReversed(false);
```

Comment:
it is bad practise to make anonymous arrow functions in jsx.
make handler in component and pass reference here